lib/spack/spack/build_systems/lua.py

```diff
@@ -29,15 +29,12 @@ class LuaPackage(spack.package_base.PackageBase):
 
     with when("build_system=lua"):
         depends_on("lua-lang")
-        extends("lua", when="^lua")
-        with when("^lua-luajit"):
-            extends("lua-luajit")
-            depends_on("luajit")
-            depends_on("lua-luajit+lualinks")
-        with when("^lua-luajit-openresty"):
-            extends("lua-luajit-openresty")
-            depends_on("luajit")
-            depends_on("lua-luajit-openresty+lualinks")
+        with when("^[virtuals=lua-lang] lua"):
+            extends("lua")
+        with when("^[virtuals=lua-lang] lua-luajit"):
+            extends("lua-luajit+lualinks")
+        with when("^[virtuals=lua-lang] lua-luajit-openresty"):
+            extends("lua-luajit-openresty+lualinks")
```

Comment:
This is certainly more correct than what was encoded before. Now we select based on the `lua-lang` provider and we won't leave any arbitrary choice to the concretizer.

Reply:
Yeah, I remember worrying about this when I set it up.  IIRC we didn't have the `^[]` provider syntax yet and this was the closest I could figure out.  :+1: from me